packages/graphql/src/schema-model/annotation/AuthorizationAnnotation.test.ts

```diff
@@ -0,0 +1,113 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { AuthorizationValidateRuleConstructor } from "./AuthorizationAnnotation";
+import {
+    AuthorizationAnnotation,
+    AuthorizationFilterOperationRule,
+    AuthorizationFilterRule,
+    AuthorizationValidateOperationRule,
+    AuthorizationValidateRule,
+} from "./AuthorizationAnnotation";
+
+describe("AuthorizationAnnotation", () => {
+    it("initialize class correctly", () => {
+        const filterRule = {
+            where: { node: { id: { gt: 1 } } },
+        };
+        const validateRule = {
+            where: { node: { name: { equals: "Keanu" } } },
+        };
+        const authFilterRule = new AuthorizationFilterRule(filterRule);
+        const authPreValidationRule = new AuthorizationValidateRule(validateRule);
+        const authAnnotation = new AuthorizationAnnotation({
+            filter: [authFilterRule],
+            validate: [authPreValidationRule],
+        });
+        expect(authAnnotation.name).toBe("AUTHORIZATION");
+        expect(authAnnotation.filter).toHaveLength(1);
+        expect(authAnnotation.filter).toEqual([
+            {
+                operations: AuthorizationFilterOperationRule,
+                requireAuthentication: true,
+                where: {
+                    jwtPayload: undefined,
+                    node: filterRule.where.node,
+                },
+            },
+        ]);
+        expect(authAnnotation.validate).toHaveLength(1);
+        expect(authAnnotation.validate).toEqual([
+            {
+                operations: AuthorizationValidateOperationRule,
+                when: ["BEFORE", "AFTER"],
+                requireAuthentication: true,
+                where: {
+                    jwtPayload: undefined,
+                    node: validateRule.where.node,
+                },
+            },
+        ]);
+    });
+});
+
+describe("AuthorizationFilterRule", () => {
+    it("initialize class correctly", () => {
+        const rule = {
+            where: { node: { name: { equals: "Keanu" } } },
+        };
+        const authFilterRule = new AuthorizationFilterRule(rule);
+        expect(authFilterRule.operations).toEqual([
+            "READ",
+            "UPDATE",
+            "DELETE",
+            "CREATE_RELATIONSHIP",
+            "DELETE_RELATIONSHIP",
+        ]);
+        expect(authFilterRule.requireAuthentication).toBeTrue();
+        expect(authFilterRule.where).toEqual({
+            jwtPayload: undefined,
+            node: rule.where.node,
+        });
+    });
+});
+
+describe("AuthorizationValidateRule", () => {
+    it("initialize class correctly", () => {
+        const rule = {
+            where: { node: { name: { equals: "Keanu" } } },
+            when: ["BEFORE"],
+        } as AuthorizationValidateRuleConstructor;
+        const authValidateRule = new AuthorizationValidateRule(rule);
```

Comment:
Do we need these extra classes? I have the feeling that we could pass the validateRule into the annotation directly and parse them there instead of creating a rule class

Reply:
At this point no, we started the design with it and then never change back, but potentially in the future they could serve some purpose. I would say there is no harm to keep it as it!